src/transactions/feed/queryHelper.ts

```diff
@@ -66,34 +63,41 @@ export function useFetchTransactions(): QueryHookResult {
   const address = useSelector(walletAddressSelector)
   const localCurrencyCode = useSelector(getLocalCurrencyCode)
 
-  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  // track cumulative transactions and most recent page info in one state, so
+  // that they do not become out of sync
+  const [fetchedResult, setFetchedResult] = useState<{
+    transactions: TokenTransaction[]
+    pageInfo: PageInfo | null
+  }>({
+    transactions: [],
+    pageInfo: null,
+  })
   const [fetchingMoreTransactions, setFetchingMoreTransactions] = useState(false)
-  const { transactions, addTransactions } = useDeduplicatedTransactions()
 
   // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
   const [counter, setCounter] = useState(0)
   useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
 
-  const handleResult = (result: QueryResponse, paginatedResult: boolean) => {
-    Logger.info(TAG, `Fetched ${paginatedResult ? 'next page' : 'new'} transactions`)
+  const handleResult = (result: QueryResponse, hasPreviousPage: boolean) => {
+    Logger.info(TAG, `Fetched ${hasPreviousPage ? 'next page' : 'new'} transactions`)
 
     const returnedTransactions = result.data?.tokenTransactionsV2?.transactions
+    const returnedPageInfo = result.data?.tokenTransactionsV2?.pageInfo
 
     if (returnedTransactions?.length) {
-      const nonEmptyTransactions = returnedTransactions.filter(
-        (returnedTransaction) => !isEmpty(returnedTransaction)
-      )
-      addTransactions(nonEmptyTransactions)
-      // We store non-paginated results in redux to show them to the users when they open the app.
-      if (!paginatedResult) {
+      setFetchedResult((prev) => ({
+        transactions: deduplicateTransactions(prev.transactions, returnedTransactions),
+        pageInfo: returnedPageInfo ?? null,
```

Comment:
@dievazqu this is one part of the logic that i changed (compared to line 94 from the previous version), can you think of any places where this condition is not okay? for the few cases i tested, this works okay

Reply:
This seems ok!